src/locale/localed.c

```diff
@@ -377,6 +380,16 @@ static int method_set_locale(sd_bus_message *m, void *userdata, sd_bus_error *er
         for (p = 0; p < _VARIABLE_LC_MAX; p++)
                 free_and_replace(c->locale[p], new_locale[p]);
 
+        hook_args[0] = NULL;
+        hook_args[1] = c->locale[VARIABLE_LANG]; /* pass the locale name we change to as first parameter to all hooks */
+        hook_args[1] = NULL;
```

Comment:
Eww, sorry for that - in my defense, it was pretty late that day and I used this pattern because I found it elsewhere in systemd's code, and because it makes it easy to keep logical blocks together.
In this case though, your suggestion is much better.